x/auction/keeper/keeper.go

```diff
@@ -0,0 +1,497 @@
+func (k Keeper) GetActiveAuctionByID(ctx sdk.Context, id uint32) (types.Auction, bool) {
+	store := ctx.KVStore(k.storeKey)
+
+	bz := store.Get(types.GetActiveAuctionKey(id))
+	if len(bz) != 0 {
+		return types.Auction{}, false
+	}
+
+	var auction types.Auction
+	k.cdc.MustUnmarshal(bz, &auction)
+	return auction, true
+}
+
+// DeleteActiveAuction deletes the active auction
+func (k Keeper) deleteActiveAuction(ctx sdk.Context, id uint32) {
+	ctx.KVStore(k.storeKey).Delete(types.GetActiveAuctionKey(id))
+}
+
+// GetEndedAuctionByID returns a specific active auction
+func (k Keeper) GetEndedAuctionByID(ctx sdk.Context, id uint32) (types.Auction, bool) {
+	store := ctx.KVStore(k.storeKey)
+
+	bz := store.Get(types.GetEndedAuctionKey(id))
+	if len(bz) != 0 {
+		return types.Auction{}, false
+	}
+
+	var auction types.Auction
+	k.cdc.MustUnmarshal(bz, &auction)
+	return auction, true
+}
+
+// IterateAuctions iterates over all auctions in the store for a given prefix
+// auctionTypePrefix for specifying whether we are iterating over active or ended auctions
+func (k Keeper) IterateAuctions(ctx sdk.Context, auctionTypePrefix []byte, handler func(auctionID uint32, auction types.Auction) (stop bool)) {
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, auctionTypePrefix)
+	defer iter.Close()
+	for ; iter.Valid(); iter.Next() {
+		key := bytes.NewBuffer(iter.Key())
+		key.Next(1) // trim prefix byte
+
+		auctionID := binary.BigEndian.Uint32(key.Bytes())
+
+		var auction types.Auction
+		k.cdc.MustUnmarshal(iter.Value(), &auction)
+		if handler(auctionID, auction) {
+			break
+		}
+	}
+}
+
+// GetActiveAuctions returns all active auctions
+func (k Keeper) GetActiveAuctions(ctx sdk.Context) []*types.Auction {
+	var auctions []*types.Auction
+	k.IterateAuctions(ctx, types.GetActiveAuctionsPrefix(), func(auctionID uint32, auction types.Auction) (stop bool) {
+		auctions = append(auctions, &auction)
+		return false
+	})
+
+	return auctions
+}
+
+// GetEndedAuctions returns all inactive auctions (that have not been pruned)
+func (k Keeper) GetEndedAuctions(ctx sdk.Context) []*types.Auction {
+	var auctions []*types.Auction
+	k.IterateAuctions(ctx, types.GetEndedAuctionsPrefix(), func(auctionID uint32, auction types.Auction) (stop bool) {
+		auctions = append(auctions, &auction)
+		return false
+	})
+
+	return auctions
+}
+
+// SetActiveAuction sets the auction specified
+func (k Keeper) setActiveAuction(ctx sdk.Context, auction types.Auction) {
+	bz := k.cdc.MustMarshal(&auction)
+	ctx.KVStore(k.storeKey).Set(types.GetActiveAuctionKey(auction.Id), bz)
+}
+
+// SetEndedAuction sets the auction specified
+func (k Keeper) setEndedAuction(ctx sdk.Context, auction types.Auction) {
+	bz := k.cdc.MustMarshal(&auction)
+	ctx.KVStore(k.storeKey).Set(types.GetEndedAuctionKey(auction.Id), bz)
+}
+
+// BeginAuction starts a new auction for a single denomination
+func (k Keeper) BeginAuction(ctx sdk.Context,
```

Comment:
Is this function used anywhere?

Reply:
Not yet, this is defined for cellar fees to interface with this module to create a new auction.